Kernel/Devices/Generic/PCSpeakerDevice.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2023, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/API/ToneGenerationPacket.h>
+#if ARCH(X86_64)
+#    include <Kernel/Arch/x86_64/PCSpeaker.h>
+#endif
+#include <Kernel/Boot/CommandLine.h>
+#include <Kernel/Devices/DeviceManagement.h>
+#include <Kernel/Devices/Generic/PCSpeakerDevice.h>
+#include <Kernel/Sections.h>
+
+namespace Kernel {
+
+UNMAP_AFTER_INIT NonnullRefPtr<PCSpeakerDevice> PCSpeakerDevice::must_create()
+{
+    auto device = MUST(DeviceManagement::try_create_device<PCSpeakerDevice>());
+    return *device;
+}
+
+UNMAP_AFTER_INIT PCSpeakerDevice::PCSpeakerDevice()
+    : CharacterDevice(1, 10)
+{
+}
+
+UNMAP_AFTER_INIT PCSpeakerDevice::~PCSpeakerDevice() = default;
+
+bool PCSpeakerDevice::can_read(OpenFileDescription const&, u64) const
+{
+    return true;
```

Comment:
Shouldn't this be `false`?

Reply:
But then you would just block forever, instead of getting immediate response that we don't support this operation?